presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestReorderWindows.java

```diff
@@ -252,7 +252,37 @@ public void testNotReorderAcrossFilter()
                                                 window(windowMatcherBuilder -> windowMatcherBuilder
                                                                 .specification(windowApp)
                                                                 .addFunction(functionCall("avg", commonFrame, ImmutableList.of(DISCOUNT_ALIAS))),
-                                                        LINEITEM_TABLESCAN_DOQRST)))))); // should be anyTree(LINEITEM_TABLESCAN_DOQRST) but anyTree does not handle zero nodes case correctly
+                                                        LINEITEM_TABLESCAN_DOQRST)))))); // should be anyTree(LINEITEM_TABLESCAN_DOQPRSST) but anyTree does not handle zero nodes case correctly
+    }
+
+    @Test
+    public void testReorderAcrossPartitionFilter()
+    {
+        @Language("SQL") String sql = "" +
+                "SELECT " +
+                "  avg_discount_APP, " +
+                "  AVG(quantity) OVER(PARTITION BY suppkey ORDER BY orderkey ASC NULLS LAST) avg_quantity_A " +
+                "FROM ( " +
+                "   SELECT " +
+                "     *, " +
+                "     AVG(discount) OVER(PARTITION BY suppkey, tax ORDER BY receiptdate ASC NULLS LAST) avg_discount_APP " +
+                "   FROM lineitem) " +
+                "WHERE suppkey > 0";
+
+        // We use all optimizers because this behavior depends on PredicatePushDown occurring before reordering of Windows
```

Comment:
You should rather add `PredicatePushDown` to `assertUnitPlan` or make
`assertUnitPlan` accept `boolean` parameter whether `PredicatePushDown` optimizer should be added